content/blogposts/2022/shift-left-wtf-does-it-mean.md

```diff
@@ -0,0 +1,184 @@
+
+Context acquisition is one of the most time-consuming parts of the job—unless you're in a completely new codebase, you probably spend more time acquiring context than you do writing new code. It's like that old saying: "If I had an hour to chop down a tree, I'd spend the first 45 minutes sharpening my axe." So it goes with context. With the proper context, you can code at the speed of thought. Without the proper context, you'll find yourself pulled out of [flow](developer-productivity-thoughts#picture-2-reaching-flow-state) again and again when you need to go look something up or realize you wasted an hour writing something that already exists.
+
+Acquiring context means understanding the structure and relationships in code. This becomes 10x easier when you have code navigation. Not having jump-to-def and xrefs is like trying to browse the web without hyperlinks.
+
+<figcaption>Walking the forward and backward graph of code (defs and refs) is the bread-and-butter of building up a contextual mental model of how the code works.</figcaption>
+<video className="blog-image" title="Definitions and references" alt="Walking the forward and backward graph of code (defs and refs) is the bread-and-butter of building up a contextual mental model of how the code works." loop autoPlay muted playsInline>
+  <source src="https://storage.googleapis.com/sourcegraph-assets/blog/shift-left/03-defs-and-refs.webm" type="video/webm" data-cookieconsent="ignore"/>
+</video>
+
+<figcaption>Fast and accurate jump-to-definition that Just Works, even across dependency boundaries, is essential for chasing down context through the winding rabbitholes of code.</figcaption>
+<video className="blog-image" title="Jump to definition" alt="Fast and accurate jump-to-definition that Just Works, even across dependency boundaries, is essential for chasing down context through the winding rabbitholes of code." loop autoPlay muted playsInline>
+  <source src="https://storage.googleapis.com/sourcegraph-assets/blog/shift-left/04-jump-to-definition.webm" type="video/webm" data-cookieconsent="ignore"/>
+</video>
+
+Targeted pieces of documentation can help guide new devs to quickly acquire the essential aspects of an area of code. You'll ideally want to use a tool that integrates nicely with the source code and doesn't go stale with time.
+
+<figcaption>Interactive docs that tie high-level descriptions with entrypoints into the source code conserves senior engineers' time while enabling other engineers to quickly onboard to new parts of the codebase.</figcaption>
+<video className="blog-image" title="Interactive docs for engineering" alt="Interactive docs that tie high-level descriptions with entrypoints into the source code conserves senior engineers' time while enabling other engineers to quickly onboard to new parts of the codebase." loop autoPlay muted playsInline>
```

Comment:
I noticed the video elements, figcaptions, and the YouTube video protrude past the content width. Did we want to match these elements to the same size as the copy/content or are they bigger on purpose? We could potentially use the `max-w-650` class which is pretty close or update our styling so it's consistent every time we make a blogpost.

I also noticed the `blog-image` class doesn't seem to do anything different on video elements after removing it, but feel free to double check my eyes.

Reply:
Ah, Brett you're right on both counts. The `blog-image` class does not appear to add anything noticeable here, and I agree that we should make the videos conform with the width of the content. 8af58c5

With a closer look, there are other instances of videos at this size on other blogposts. This may warrant a batch change on other blogposts. Although we are moving away from writing special classes, we could use alternatively use `max-width: 650px` on `blog-image`, which is assigned to many of these extra large videos.

Reply:
@st0nebraker and I noticed that the embedded videos don't play on Safari. I suppose WEBM is still causing Safari woes. To ensure we have coverage, lets request MP4 versions from Fabiana and Nick to incorporate as backups. 

Lmk if you have any thoughts on this, @bretthayes!

Reply:
You could update `_BlogPost.scss`, select `video, figcaption, iframe` and add a `max-width` in the mean time until we can circle back and do a batch change. The blog needs some styling love soon.

Also, you can use `ffmpeg` if you need to create an mp4 from the webm if it's going to take too long to get mp4s.

```sh
brew install ffmpeg
ffmpeg -i input.webm -vcodec h264 -b:v 600k output.mp4
```

You can play with the kbits or use different flags if you want to meet a specific file size, resolution, or bitrate. 

Check out the [browser usage for webm](https://caniuse.com/?search=webm).
> "Safari 14.1 – 15.6 has full support of WebM, but requires macOS 11.3 Big Sur or later."